src/deploydocs.jl

```diff
@@ -268,6 +268,14 @@ function deploydocs(;
                 @debug "running extra build steps."
                 make()
             end
+            objects_inv = joinpath(realpath(target), "objects.inv")
+            if isfile(objects_inv)
+                inventory_version = _get_inventory_version(objects_inv)
+                deploy_version = _get_deploy_version(deploy_subfolder)
+                if !isempty(deploy_version) && (inventory_version != deploy_version)
+                    error("Inventory declares version `$inventory_version`, but `deploydocs` is for version `$deploy_version`")
+                end
+            end
```

Comment:
@mortenpi This is the bit of code in `deploydocs` that would guarantee that there is never a mismatch between the automatic version extracted from the `Project.toml` and what is shown in the versions-selection menu for tagged releases.

Reply:
If you think the `error` is too harsh or might be considered semver-breaking, I'd also be okay making it an `@error`, and potentially even overwriting the `inventory_version` with the `deploy_version`. That would be *extremely* close to #2448, but the "fallback" would happen in `makedocs` instead of `deploydocs`, which I'd much prefer.

Reply:
See #2449, which might be starting to become my *preferred* solution (since I actually agree with you that for a tagged release, the `version` that comes out of `deploydocs` is kinda authoritative).